From be3b2f227a6cc60e27f2b74e9f301bcc136183e6 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Wed, 29 Oct 2025 08:26:50 +0100
Subject: [PATCH 1/2] Explicitely link with libpython on Cygwin

Like with Windows Python, under Cygwin we need to explicitely
link with libpython.

Thix fixes the build under Cygwin.
---
 pyo3-build-config/src/impl_.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyo3-build-config/src/impl_.rs b/pyo3-build-config/src/impl_.rs
index cca9393d..b977ac79 100644
--- a/pyo3-build-config/src/impl_.rs
+++ b/pyo3-build-config/src/impl_.rs
@@ -890,6 +890,7 @@ pub fn is_linking_libpython_for_target(target: &Triple) -> bool {
         || target.operating_system == OperatingSystem::Aix
         || target.environment == Environment::Android
         || target.environment == Environment::Androideabi
+        || target.operating_system == OperatingSystem::Cygwin
         || !is_extension_module()
 }
 
-- 
2.51.2

